Adding a new scenario as plan B (#51)
odh-labs/hackathon: data/hackathon/scenario7.mdx

---

> title: Accelerate GenAI experimentation by using an open source Enterprise-grade tool and software supply chain to get started

Reviewer: Keep the titles short imo

```suggestion
title: Accelerate GenAI experimentation with Podman
```

exercise: 7

> date: '2024-06-25'

Reviewer: Update date

```suggestion
date: '2024-07-02'
```

tags: ['podman desktop','ai lab','genai', 'granite','ai applications']
draft: false
authors: ['default']

> summary: "Demonstrate Podman Desktops in-built"

Reviewer: ```suggestion
summary: "Can you pull of a clutch live demo for Podman Desktops new AI features?"
```

---

> ACME loved the flexibility at enterprise grade quality and scale of OpenShift AI and also the game changing approach to LLMs through InstructLab. 
> But all this excitement needs to be pointed in to the right direction with the right tooling right from when experimentation starts.
> "Can you help us?" they want to know - "I am glad you asked" is what you are saying.

Reviewer: Adding some extra flavour in the scenario

```suggestion
The ACME Data Science team loved the flexibility, scale and enterprise grade quality of OpenShift AI. But all this excitement needs to be pointed in to the right direction with the right tooling right from when experimentation starts.

The ACME Senior Manager for Data Science Dave McDataScienceFace is concerned their data science team are still running ancient versions of docker desktop (`4.0.1` from 2021 🤮) and that every data scientist in the team seems to be spending weeks just getting basic local AI development environments running on their laptops.

The team workflows heavily rely on local docker containers for development so this seems like a great opportunity to push for ACME Financial Services to adopt 🌟 [Podman Desktop with Podman AI Lab](https://developers.redhat.com/articles/2024/05/07/introducing-podman-ai-lab) 🌟.

You've agreed to put together a quick demo to run for Dave and the Data Scientists at their upcoming team meeting. Can you secure agreement for the adoption of Podman Desktop? 

If you can pull this off Dave McDataScienceFace reckons this will accelerate how quickly the data science team are consuming ROSA compute time by getting models developed & deployed to OpenShift AI much more quickly.
```

1) Get Podman Desktop running on  your laptop.
2) Install the AI Lab extension
3) Use the AI Lab extension to build your demo app and to access to your code
4) Utilize the Recipe catalogue to deploy the Summarizer recipe
5) Change the application code so that it says 'Hello from Team <Your team name>!'
6) Take a pdf document and let the Summarizer app summarize it.
7) Take a screenshot and show that you changed the Greeting as well as that the summarizer app ran successfully.

## 7.0 - Be in the Know...
Documentation you might find useful
- https://podman-desktop.io/extensions/ai-lab
- https://developers.redhat.com/articles/2024/05/07/podman-ai-lab-getting-started?source=sso#catalog_of_open_source_models

## 7.1 - Setting Up
The ACMEFS team wants to see how you install Podman Desktop and the AI Lab extension on your laptop
- Demonstrate the installation process or upgrade process of Podman Desktop.

## 7.2 - Introducing Podman
Explain what you see when going through podman desktop.
- The team wants to know what version of Podman is required to install the AILab extension.
- Show the ACME team how you can view and access containers, images and volumes

> - Browse the current models installed and show how you can download a new model.

Reviewer: Let's include a screenshot here of Podman AI to make the page a little more interesting.

## 7.3 - Text Summarizer
Demonstrate the ACME team how you have out of the box recipes you can leverage.
- Use the Summarizer recipe, change the code such that it shows a welcome message with your team name in it.
- Take a screenshot and post into the slack channel.

# HINTS

> If you get stuck on a question, fear not, perhaps try a different approach. If you have tried everything you can think of and are still stuck you can unlock a hint for `5` points by posting a message in the `#event-anz-ocp-ai-hackathon` channel with the message:

Author: Thank you!
